sys/net/gnrc/transport_layer/tcp/gnrc_tcp_pkt.c

```diff
@@ -0,0 +1,434 @@
+/*
+ * Copyright (C) 2015 Simon Brummer
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gnrc
+ * @{
+ *
+ * @file
+ * @brief       GNRC's TCP paket related functions
+ *
+ * @author      Simon Brummer <brummer.simon@googlemail.com>
+ * @}
+ */
+#include <stdlib.h>
+#include <utlist.h>
+#include <errno.h>
+#include "msg.h"
+#include "net/gnrc/pktbuf.h"
+#include "net/gnrc/tcp.h"
+#include "tcp_internal/pkt.h"
+#include "tcp_internal/helper.h"
+#include "tcp_internal/option.h"
+#include "tcp_internal/eventloop.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/* Maximum Macro */
+#define MAX(x, y) (((x) > (y)) ? (x) : (y))
```

Comment:
Please use inline functions.

Also I think there is a central function for this somewhere.